playbooks/robusta_playbooks/api_service.py

```diff
@@ -0,0 +1,139 @@
+import json
+import logging
+from typing import List, Optional
+
+import yaml
+from kubernetes.client import ApiregistrationV1Api, V1APIServiceList
+from pydantic import BaseModel
+from robusta.api import BaseBlock, FileBlock, MarkdownBlock, PrometheusKubernetesAlert, action
+
+
+class ApiServiceConditions(BaseModel):
+    message: Optional[str]
+    reason: Optional[str]
+    status: Optional[str]
+    type: Optional[str]
+
+
+class ApiService(BaseModel):
+    name: str
+    service_name: Optional[str]
+    service_namespace: Optional[str]
+    conditions: List[ApiServiceConditions]
+    full_json: dict
+
+    def get_json(self):
+        return self.full_json
+
+    def get_yaml(self):
+        # remove noisy fields
+        if self.full_json.get("metadata", {}).get("managedFields", {}):
+            del self.full_json["metadata"]["managedFields"]
+        last_applied = "kubectl.kubernetes.io/last-applied-configuration"
+        if self.full_json.get("metadata", {}).get("annotations", {}).get(last_applied):
+            del self.full_json["metadata"]["annotations"][last_applied]
+        return yaml.dump(self.full_json)
+
+    def get_available_condition(self) -> Optional[ApiServiceConditions]:
+        if not self.conditions:
+            return None
+        available_conditions = [condition for condition in self.conditions if condition.type == "Available"]
+        if not available_conditions:
+            return None
+        return available_conditions[0]
+
+    @staticmethod
+    def parse_apiservice(apiservice: dict, name: str):
+        service = apiservice.get("spec", {}).get("service", {})
+        service_name = service.get("name")
+        namespace = service.get("namespace")
+        conditions = apiservice.get("status", {}).get("conditions", [])
+        return ApiService(
+            name=name,
+            service_name=service_name,
+            service_namespace=namespace,
+            conditions=conditions,
+            full_json=apiservice,
+        )
+
+
+def get_api_service(name: str) -> Optional[ApiService]:
+    try:
+        field_selector = f"metadata.name={name}"
+        api_services_json: V1APIServiceList = (
+            ApiregistrationV1Api()
+            .list_api_service(field_selector=field_selector, _preload_content=False)
+            .data.decode("utf-8")
+        )
+        api_services_json_list = json.loads(api_services_json).get("items", [])
+        if len(api_services_json_list) == 0:
+            logging.info(f"Api service {name} not found.")
+            return None
+        if len(api_services_json_list) > 1:
```

Comment:
I am not sure if its a real use case or more of a sanity check. 
If its possible ( 2 services that expose the same api service? ) it might be necessary to handle that case.
But I am not sure if this case is even possible

Reply:
Shouldn't be possible but its a sanity check since list_api_service has issues and is an old version.